### PR TITLE
Refactor FXIOS-8388 [v124] Move ActionFooterView to Fakespot code

### PR DESCRIPTION
**Pull request by bmihai23**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8388)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18594)

## :bulb: Description
- This PR moves the ActionFooterView, currently part of the ComponentLibrary, to where it's used in Client/Frontend/Fakespot/Views. 
- This change improves code organization and avoids treating it as a separate component. 

### Changes Made
- Renamed ActionFooterView to FakespotActionFooterView
- Renamed ActionFooterViewModel to FakespotActionFooterViewModel
- Moved FakespotActionFooterView to Client/Frontend/Fakespot/Views
- Removed documentation for ActionFooterView from ComponentLibrary
- Removed ActionFooterComponentViewModel and ActionFooterViewController from SampleComponentLibraryApp

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift [hunk 1]
```diff
@@ -44,11 +45,10 @@ public final class ActionFooterView: UIView, ThemeApplicable {
         label.adjustsFontForContentSizeCategory = true
     }
 
-    private lazy var linkButton: LegacyResizableButton = .build { button in
+    private lazy var linkButton: ResizableButton = .build { button in
```
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift:48]: This change plus actually causes UI issues where the button is either too big or displayed in with lots of space above. Let's undo this change and the removal of the buttonEdgeSpacing a few lines down and address the deprecation warning in another PR. I'll create a task for it.
Simulator Screenshot - iPhone 15 Pro Max - 2024-02-07 at 09 35 26
Simulator Screenshot - iPhone 15 Pro Max - 2024-02-07 at 09 35 18

**Review by thatswinnie — COMMENTED** [review]
Thanks for grabbing this task @bmihai23 ! 🙏 

Unfortunately the SampleComponentLibraryApp doesn't build anymore because the files were not removed from the [SampleComponentLibraryApp Xcode project](https://github.com/mozilla-mobile/firefox-ios/tree/main/SampleComponentLibraryApp/SampleComponentLibraryApp.xcodeproj) correctly. Could you please get this fixed?

**Comment by bmihai23** [comment]
> Thanks for grabbing this task @bmihai23 ! 🙏
> 
> Unfortunately the SampleComponentLibraryApp doesn't build anymore because the files were not removed from the [SampleComponentLibraryApp Xcode project](https://github.com/mozilla-mobile/firefox-ios/tree/main/SampleComponentLibraryApp/SampleComponentLibraryApp.xcodeproj) correctly. Could you please get this fixed?

Thank you for your feedback, @thatswinnie! I've addressed the issue and fixed it.

**Review by thatswinnie — APPROVED** [review]
Perfect. Thank you so much @bmihai23 🙏

File: firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift (hunk 1 above)
Comment (thatswinnie) [reply firefox-ios/Client/Frontend/Fakespot/Views/FakespotActionFooterView.swift]: The task to fix the deprecation warnings for the button: https://github.com/mozilla-mobile/firefox-ios/issues/18634

**Pull request merged**